### PR TITLE
Ticket 1: Signup / Login Pages

### PR DESCRIPTION
**Pull request by samsoncsyu7777**
### What this PR does (required):
- Sign up and Login header buttons linked
- Sign up and Login text linked
- Styling as mockups

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

Screenshot 2021-11-10 at 5 48 12 AM

Screenshot 2021-11-10 at 5 48 07 AM

### Any information needed to test this feature (required):
- Sign up as a new user
- Click Sign up button or text to route to the Sign up page
- Click Login button or text to route to the Login page

### CODE REVIEW

**Comment by rajivtitus** [comment]
Hey @samsoncsyu7777 if this PR is ready for review, please add me as a reviewer :)

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 1]
```diff
@@ -3,25 +3,36 @@ import Box from '@material-ui/core/Box';
...
-      </Link>
-    </Box>
+    <LogoHeader>
+      <Box p={1} className={classes.authHeader}>
+        <Link to={becomeSitterLink} className={classes.link}>
+          <Typography className={classes.accAside}>BECOME A SITTER</Typography>
+        </Link>
+        <Link to={loginLink} className={classes.link}>
+          <Button color="secondary" className={classes.accBtn} variant={btnText === 'LOGIN' ? 'contained' : 'outlined'}>
+            LOGIN
```
Comment (sundayezeilo) [inline client/src/components/AuthHeader/AuthHeader.tsx]: Please avoid using CAPITALCASE for text, some screen readers interpret this text as abbreviations, you can resort to using a CSS class to transform text instead.

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 2]
```diff
@@ -3,25 +3,36 @@ import Box from '@material-ui/core/Box';
...
+          </Button>
+        </Link>
+        <Link to={signupLink} className={classes.link}>
+          <Button color="secondary" className={classes.accBtn} variant={btnText === 'LOGIN' ? 'outlined' : 'contained'}>
+            SIGN UP
```
Comment (sundayezeilo) [inline client/src/components/AuthHeader/AuthHeader.tsx]: The same with this one

File: client/src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -39,14 +39,14 @@ export default function Login(): JSX.Element {
...
+      <AuthHeader btnText="LOGIN" />
+      <Grid container xs={12} sm={8} md={7} className={classes.elevationGrid} component={Paper} square>
         <Box className={classes.authWrapper}>
-          <AuthHeader linkTo="/signup" asideText="Don't have an account?" btnText="Create account" />
           <Box width="100%" maxWidth={450} p={3} alignSelf="center">
             <Grid container>
               <Grid item xs>
                 <Typography className={classes.welcome} component="h1" variant="h5">
-                  Welcome back!
+                  Login
```
Comment (sundayezeilo) [inline client/src/pages/Login/Login.tsx]: The same comment here with CAPITALCASE

File: client/src/pages/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -40,14 +40,14 @@ export default function Register(): JSX.Element {
...
+      <AuthHeader btnText="SIGN UP" />
+      <Grid container xs={12} sm={8} md={7} className={classes.elevationGrid} component={Paper} square>
         <Box className={classes.authWrapper}>
-          <AuthHeader linkTo="/login" asideText="Already have an account?" btnText="Login" />
           <Box width="100%" maxWidth={450} p={3} alignSelf="center">
             <Grid container>
               <Grid item xs>
                 <Typography className={classes.welcome} component="h1" variant="h5">
-                  Create an account
+                  Sign up
```
Comment (sundayezeilo) [inline client/src/pages/SignUp/SignUp.tsx]: :+1:

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 5]
```diff
@@ -51,67 +53,81 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
...
             error={touched.password && Boolean(errors.password)}
             value={values.password}
             onChange={handleChange}
           />
-
           <Box textAlign="center">
-            <Button type="submit" size="large" variant="contained" color="primary" className={classes.submit}>
-              {isSubmitting ? <CircularProgress style={{ color: 'white' }} /> : 'Create'}
+            <Button type="submit" size="large" variant="contained" color="secondary" className={classes.submit}>
+              {isSubmitting ? <CircularProgress style={{ color: 'white' }} /> : 'SIGN UP'}
```
Comment (sundayezeilo) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: Please avoid using CAPITALCASE for text, some screen readers interpret this text as abbreviations, you can resort to using a CSS class to transform text instead.

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 6]
```diff
@@ -51,67 +53,81 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
...
             </Button>
           </Box>
+          <Box textAlign="center" className={classes.loginBox}>
+            <Typography className={classes.isMember}>Already a member?</Typography>
+            <Link to={loginLink} className={classes.link}>
+              <Typography className={classes.loginLink}>Login</Typography>
```
Comment (sundayezeilo) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: :+1:

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
UI looks good! I left a few comments on the places you used CAPITALCASE for text. Please address those comments. Also, it is required that you link the issue you are addressing in each of your PRs. See [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

**Comment by samsoncsyu7777** [comment]
Hi Sunday,

Thanks a lot for your comments. May I know more about the ticket 2? Is it
styling the navbar and make the skeleton for each page?

Best regards,
Charm

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 7]
```diff
@@ -3,25 +3,36 @@ import Box from '@material-ui/core/Box';
...
-      <Link to={linkTo} className={classes.link}>
-        <Button color="inherit" className={classes.accBtn} variant="contained">
-          {btnText}
-        </Button>
-      </Link>
-    </Box>
+    <LogoHeader>
+      <Box p={1} className={classes.authHeader}>
+        <Link to={becomeSitterLink} className={classes.link}>
+          <Typography className={classes.accAside}>Become a sitter</Typography>
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/AuthHeader.tsx]:
```suggestion
          <Typography component={Link} to="/become-sitter" className={classes.accAside}>Become a sitter</Typography>
```
The react router Link and MUI's Typography can be combined :)

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 8]
```diff
@@ -3,25 +3,36 @@ import Box from '@material-ui/core/Box';
...
+        </Link>
+        <Link to={loginLink} className={classes.link}>
+          <Button color="secondary" className={classes.accBtn} variant={btnText === 'login' ? 'contained' : 'outlined'}>
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/AuthHeader.tsx:25]: Same as comment above, the MUI Button can be combined with Link

File: client/src/components/AuthHeader/useStyles.ts [hunk 9]
```diff
@@ -1,17 +1,23 @@
 import { makeStyles } from '@material-ui/core/styles';
+import { custom } from '../../themes/custom';
 
 const useStyles = makeStyles((theme) => ({
   authHeader: {
+    height: '100%',
     alignSelf: 'flex-end',
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/useStyles.ts]: Most of these styles can be applied to the Box component :)

Eg: `<Box display="flex" alignItems="center"....>`

File: client/src/components/LogoHeader/useStyles.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    width: '100%',
+    height: 70,
+    borderRadius: theme.shape.borderRadius,
+    filter: 'drop-shadow(0px 2px 6px rgba(74,106,149,0.2))',
+    backgroundColor: custom.colors.white,
+    boxShadow: 'none',
+  },
+  logoBox: {
+    height: '100%',
+    alignSelf: 'flex-start',
```
Comment (rajivtitus) [inline client/src/components/LogoHeader/useStyles.ts]: Most of these styles can be applied to the Box component too!

File: client/src/pages/Login/Login.tsx [hunk 11]
```diff
@@ -22,6 +22,7 @@ export default function Login(): JSX.Element {
   ) => {
     login(email, password).then((data) => {
       if (data.error) {
+        console.error({ error: data.error.message });
```
Comment (rajivtitus) [inline client/src/pages/Login/Login.tsx]: please remove console.log

File: client/src/pages/SignUp/SignUpForm/useStyles.ts [hunk 12]
```diff
@@ -1,35 +1,57 @@
...
     fontWeight: 'bold',
+    textTransform: 'uppercase',
+    boxShadow: 'none',
   },
+  loginBox: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  isMember: {
+    fontSize: 13,
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/useStyles.ts]: Optional: You can consider defining font sizes for variants within MUI's theme and use those variants throughout the app 😄

File: client/src/pages/SignUp/useStyles.ts [hunk 13]
```diff
@@ -1,26 +1,37 @@
...
+    width: '100vw',
   },
   welcome: {
-    fontSize: 26,
-    paddingBottom: 20,
-    color: '#000000',
-    fontWeight: 700,
-    fontFamily: "'Open Sans'",
+    fontSize: 30,
+    textTransform: 'capitalize',
```
Comment (rajivtitus) [inline client/src/pages/SignUp/useStyles.ts:29]: 👍🏽

**Comment by samsoncsyu7777** [comment]
Thanks, Rajiv. May I know more details about the second ticket: Skeleton?

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 7 above)
Comment (samsoncsyu7777) [reply client/src/components/AuthHeader/AuthHeader.tsx]: Thanks for your suggestion.

**Review by rajivtitus — COMMENTED** [review]
Great work making those requested changes! Just a few comments for you to have a look :)

**Comment by rajivtitus** [comment]
> Thanks, Rajiv. May I know more details about the second ticket: Skeleton?

Your welcome Samson! You can pick any of the available tickets as the next ticket. In case you do pick the Dashboard skeleton, you will have to refactor the navbar to make sure it's persistent across the app and then conditionally render links once a user logs in (at the moment don't worry about rendering different links for different types of users). As for the content pages you can just create empty components and hook them up with the links in the navbar 😄

File: client/src/pages/SignUp/SignUpForm/useStyles.ts (hunk 12 above)
Comment (samsoncsyu7777) [reply client/src/pages/SignUp/SignUpForm/useStyles.ts]: Good suggestions. Thanks.

**Comment by samsoncsyu7777** [comment]
> > Thanks, Rajiv. May I know more details about the second ticket: Skeleton?
> 
> Your welcome Samson! You can pick any of the available tickets as the next ticket. In case you do pick the Dashboard skeleton, you will have to refactor the navbar to make sure it's persistent across the app and then conditionally render links once a user logs in (at the moment don't worry about rendering different links for different types of users). As for the content pages you can just create empty components and hook them up with the links in the navbar 😄

Many thanks for your support.

File: client/src/themes/theme.ts [hunk 14]
```diff
@@ -9,6 +9,20 @@ export const theme = createMuiTheme({
       textTransform: 'none',
       fontWeight: 700,
     },
+    h4: {
+      fontWeight: 'bold',
+      fontSize: 11,
```
Comment (rajivtitus) [inline client/src/themes/theme.ts:14]: 💯

**Review by rajivtitus — APPROVED** [review]
Excellent work implementing those changes! 💯

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 1 above)
Comment (samsoncsyu7777) [reply client/src/components/AuthHeader/AuthHeader.tsx]: Thanks. Done

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 2 above)
Comment (samsoncsyu7777) [reply client/src/components/AuthHeader/AuthHeader.tsx]: OK

File: client/src/pages/Login/Login.tsx (hunk 3 above)
Comment (samsoncsyu7777) [reply client/src/pages/Login/Login.tsx]: Done.

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx (hunk 5 above)
Comment (samsoncsyu7777) [reply client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: I see. I will remember this issue.

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx (hunk 6 above)
Comment (samsoncsyu7777) [reply client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: Thanks.

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 8 above)
Comment (samsoncsyu7777) [reply client/src/components/AuthHeader/AuthHeader.tsx:25]: Done.

File: client/src/components/AuthHeader/useStyles.ts (hunk 9 above)
Comment (samsoncsyu7777) [reply client/src/components/AuthHeader/useStyles.ts]: Done.

File: client/src/components/LogoHeader/useStyles.ts (hunk 10 above)
Comment (samsoncsyu7777) [reply client/src/components/LogoHeader/useStyles.ts]: Done.

File: client/src/pages/Login/Login.tsx (hunk 11 above)
Comment (samsoncsyu7777) [reply client/src/pages/Login/Login.tsx]: OK.

File: client/src/themes/theme.ts (hunk 14 above)
Comment (samsoncsyu7777) [reply client/src/themes/theme.ts:14]: Thanks.

**Pull request merged**